devdoc/constbuffer_array_requirements.md

```diff
@@ -139,6 +142,30 @@ MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_remove_front, CO
 
 **SRS_CONSTBUFFER_ARRAY_02_036: [** If there are any failures then `constbuffer_array_remove_front` shall fail and return `NULL`. **]**
 
+### constbuffer_array_append
+
+```c
+MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_append, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_handle, CONSTBUFFER_ARRAY_HANDLE, appended_constbuffer_array_handle);
```

Comment:
>append [](start = 64, length = 6)

So this is not really append. It is create_from_2_arrays.
I think it should be named as such.

If we are doing it as a constructor I can also suggest creating an array from several arrays (not bound it to 2, but make it generic).

Reply:
Updated with constbuffer_array_create_from_array_array that takes an array of CONSTBUFFER_ARRAY_HANDLE

---
In reply to: [245128978](https://github.com/Azure/azure-c-shared-utility/pull/248#discussion_r245128978) [](ancestors = 245128978)